ls-ui-react-app/src/pages/UploadCircuitPage.tsx

```diff
@@ -47,17 +51,28 @@ const UploadACircuit = ( {appState, setAppState} : AppStateProps ) => {
             'circuit': circuitStr,
             'apply_litinski_transform': doLitinskiTransform
         }).then( (response ) => {
-            const responseJson = JSON.parse(response.data) as CompilationResult
             setAppState({
                 ...appState,
-                compilationResult: responseJson,
-                compilationIsLoading: false
+                request: true,
+                apiResponse: response,
             })
+            try {
+                const responseJson = JSON.parse(response.data) as CompilationResult
+                setAppState({
+                    ...appState,
+                    compilationResult: responseJson,
+                })
+            }
+            // catch errors due to parsing JSON
+            catch(err) {
+                // JSON parsing error
```

Comment:
Ok I see what you were saying about parse errors, this should set an error somewhere.

Considering the 3 error cases below + this new error case, things might start to get confusing.

To keep track of everything, here is a suggestion:
Make a new datatype that clearly defines the the possible states after a response, maybe something like this:
```ts
type ApiResponse =  null | ServerError | CompilationError | ResponseParseError | CompilationResult
```
where these types are trivial classes like 
```
class ServerError {
    status: number
    errorMsg: string
}

class CompilationError {
   errorMsg: string
}
```
 then you can reduce appState to something super simple like
```ts
class AppState {
    compilationIsLoading : boolean = false
    apiResponse? :  ApiResponse
}
```
And still preserve that nice dispatch of error cases you have below in JSX

Reply:
The error/success identification and the `apiResponse` would be set in the `axios` callback

Reply:
Is there a way to pass the AXIOS response object to the .catch block?

Or is there some other way to handle the response object AFTER the JSON.parse error has been caught?